src/adaptors/hyperdrive/index.js

```diff
@@ -0,0 +1,324 @@
+const ethers = require("ethers")
+const axios = require('axios');
+const sdk = require('@defillama/sdk');
+const utils = require('../utils');
+const providers = require('@defillama/sdk/build/providers.json');
+const { GET_POOL_CONFIG_ABI, GET_POOL_INFO_ABI, POSITION_ABI, MARKET_ABI } = require('./abi');
+
+const FUTURE_REGISTRY_ADDRESS = "0x6668310631Ad5a5ac92dC9549353a5BaaE16C666"
+
+const config = {
+  ethereum: { registry: '0xbe082293b646cb619a638d29e8eff7cf2f46aa3a', },
+  xdai: { registry: '0x666fa9ef9bca174a042c4c306b23ba8ee0c59666', },
+  base: {},
+  linea: {},
+}
+
+async function queryPoolHoldings(poolContract, config, name) {
+  let baseTokenBalance;
+  let vaultSharesBalance;
+  let vaultContractAddress;
+
+  // Query base token balance
+  if (config.baseToken === '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE') {
+    // ETH case
+    baseTokenBalance = (
+      await sdk.api.eth.getBalance({
+        target: poolContract.address,
+        chain: config.chain
+      })
+    ).output;
+  } else if (name?.includes(' LP ')) {
+    // LP token case
+    const gauge_contract_address = (await sdk.api.abi.call({
+      target: poolContract.address, //"0xf49D1f422a7661541033C566f358E944a2bFb976"
+      chain: config.chain,
+      abi: 'function gauge() view returns (address)',
+    })).output;
+    baseTokenBalance = (
+      await sdk.api.erc20.balanceOf({
+        target: gauge_contract_address,
+        owner: poolContract.address,
+        chain: config.chain
+      })
+    ).output;
+  } else if (config.baseToken !== '0x0000000000000000000000000000000000000000') {
+    // Standard ERC20 case
+    baseTokenBalance = (
+      await sdk.api.erc20.balanceOf({
+        target: config.baseToken,
+        owner: poolContract.address,
+        chain: config.chain
+      })
+    ).output;
+  } else {
+    baseTokenBalance = '0';
+  }
+
+  // Query vault shares balance
+  if (name?.includes('Morpho')) {
```

Comment:
I would encapsulate this into it's own function, would help with readability

Reply:
done here, along with a few other functions d65755d16f1f7eabf69fd486f15d43705c15037e